rdr_service/tools/tool_libs/curation.py

```diff
@@ -287,7 +291,11 @@ def _get_base_src_clean_answers_select(session):
                 func.left(QuestionnaireResponseAnswer.valueString, 1024),
                 QuestionnaireResponseAnswer.valueDate,
                 QuestionnaireResponseAnswer.valueDateTime,
-                answer_code.display
+                answer_code.display,
+                case([  # Use valueInteger if the question code should be re-mapped
+                    (question_code.value.in_(zipcode_question_codes_to_remap),
```

Comment:
I may be misreading this, but shouldn't this be `notin_` since you don't want to map zip codes to `valueInteger`?

Reply:
I think this is where we'd want to check that it is one of the codes to map (rather than using `notin_`).

This block maps these QuestionnaireResponseAnswer values from `valueString`, `valueDate`, `valueDateTime`, and `display` to the SrcClean `value_string` field. If the coalesce for setting the SrcClean.value_string column finds all of those fields to be NULL, then it checks to see if the question code matches one of the values we want to map. 

The two codes in `zipcode_question_codes_to_remap` sometimes have the answer in `valueInteger` instead. So if everything else is NULL, and the `in_` check results in True. Then we want to pass QuestionnaireResponseAnswer's `valueInteger` to SrcClean's `value_string` because curation is expecting zip codes in the `value_string` field.

Let me know if that clears it up. I could be thinking about this the wrong way around too.

Reply:
Thanks, that makes sense.